mvc/src/main/java/nextstep/mvc/view/JsonView.java

```diff
@@ -1,13 +1,48 @@
 package nextstep.mvc.view;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import jakarta.servlet.http.HttpServletRequest;
 import jakarta.servlet.http.HttpServletResponse;
+import java.io.PrintWriter;
 import java.util.Map;
+import nextstep.web.support.MediaType;
 
 public class JsonView implements View {
 
+    private final ObjectMapper objectMapper;
+
+    public JsonView() {
+        this.objectMapper = new ObjectMapper();
+        objectMapper.writerWithDefaultPrettyPrinter();
+    }
+
     @Override
     public void render(final Map<String, ?> model, final HttpServletRequest request, HttpServletResponse response)
             throws Exception {
+        final String value = writeJsonData(model);
+        final PrintWriter writer = response.getWriter();
+
+        response.setContentType(MediaType.APPLICATION_JSON_UTF8_VALUE);
+        writer.write(value);
+        writer.flush();
+    }
+
+    private String writeJsonData(final Map<String, ?> model) throws JsonProcessingException {
+        if (model.isEmpty()) {
+            return "";
+        }
+        if (model.size() == 1) {
+            return objectMapper.writeValueAsString(getFirstData(model));
+        }
+        return objectMapper.writeValueAsString(model);
+    }
+
+    private Object getFirstData(final Map<String, ?> model) {
+        return model.keySet()
+                .stream()
+                .map(model::get)
+                .findFirst()
+                .orElseThrow();
```

Comment:
c: `keySet` 말고 `values`를 사용하면 더 간단하겠네요

Reply:
이거는 조금 취향차이 같긴 한데 아래 방법 말씀하시는 거 맞을까요?
```java
return model.values()
                .toArray()[KEY_INDEX] // KEY_INDEX: 0
                .toString();
```
모델의 데이터가 1개일 때이므로 성능차이는 없을 것 같고, 그럴 일은 없겠지만 만약 예외가 터진다면 stream의 경우 `orElseThrow()`로 예외처리를 할 수 있어보여서 변경의 필요성이 크게 느껴지지는 않더라구요. 일단은! 냅둬보겠습니다. 만약 제가 인지하지 못한 부분이 있다면 추가 코멘트 부탁드려요

Reply:
```suggestion
        return model.values()
                .stream()
                .findFirst()
                .orElseThrow();
```
요런 느낌을 생각했는데 크게 중요한 부분은 아닌 것 같아요~